### PR TITLE
allow multiple dashes in StatsD metric names

### PR DESCRIPTION
**Pull request by evandam**
See https://github.com/prometheus/statsd_exporter/issues/380

@matthiasr I don't do any Go development so not sure if there's anything else to consider here 😅 

Thanks!

### CODE REVIEW

**Comment by matthiasr** [comment]
There is a test that specifically said "double dashes are illegal" – it is intended to exercise _a_ case that doesn't match this regex. Could you change it to use some other illegal character?

**Comment by matthiasr** [comment]
Also, please add the DCO sign-off, instructions are behind the Details link

**Comment by SuperQ** [comment]
It might also be useful to add a positive test like this: (line 1467)

```go
    {
      testName: "Config with multipel dashes",
      config: `mappings:
- match: "*.fabio--requests.count"
  name: "fabio_requests_count"
  help: "Fabio request count"
  labels:
    hostname: "$1"`,
      mappings: mappings{
        {
          statsdMetric: "test.fabio--requests.count",
          name:         "fabio_requests_count",
          labels: map[string]string{
            "hostname": "test",
          },
        },
      },
    },              },
```

File: mapping.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+mappings:
```
Comment (matthiasr) [inline mapping.yml:1]: did you commit this file intentionally?

**Comment by matthiasr** [comment]
Something I am wondering: what happens if we accept a statsd metric with `--` but have no mapping for it? Will it be transformed to a valid Prometheus metric?

File: mapping.yml (hunk 1 above)
Comment (evandam) [reply mapping.yml]: @matthiasr you're too fast man! Adding a new test like @SuperQ recommended, committed by accident 😇

**Comment by evandam** [comment]
In my super quick testing it seems like it gets dropped if there's no mapping:

```
level=debug ts=2021-06-17T16:51:00.278Z caller=line.go:214 msg="Bad line from StatsD" line=X
level=debug ts=2021-06-17T16:51:00.290Z caller=listener.go:73 msg="Incoming line" proto=udp line=--.count:0|c
level=debug ts=2021-06-17T16:51:00.290Z caller=listener.go:73 msg="Incoming line" proto=udp line=
```

A mapping for `match: "--"` is invalid.

```
level=error ts=2021-06-17T16:52:24.339Z caller=main.go:340 msg="error loading config" error="invalid match: --"
```

It seems reasonable enough to me but it's up to you folks!

**Comment by matthiasr** [comment]
Hmm, but for metrics with a single dash there is an automatic mapping. It will be very confusing to users if _some_ metrics are mapped automatically but others are not. Do you feel up to extending the [metric name escaping](https://github.com/prometheus/statsd_exporter/blob/master/pkg/mapper/escape.go) so that it handles this input?

**Comment by evandam** [comment]
Hey @matthiasr,it sounds like you folks know better than me to get this over the line. I don't write Go either so maybe I'm not the best to handle this if there's more to it than is already done 😅

**Comment by matthiasr** [comment]
I created a new issue for handling the default case: #389.

**Review by matthiasr — APPROVED** [review]
Thank you, and sorry for holding it up for so long!

**Pull request merged**